scanapp/models.py

```diff
@@ -31,86 +31,55 @@
 from rest_framework.authtoken.models import Token
 
 
-# Create your models here.
 @receiver(post_save, sender=settings.AUTH_USER_MODEL)
 def create_auth_token(sender, instance=None, created=False, **kwargs):
     if created:
         Token.objects.create(user=instance)
 
 
-class ScanInfo(models.Model):
+class Scan(models.Model):
+    """
+    Store various attributes of a scan like `scan_type`, `is_complete`, `user`, 
+    `URL`, `code_directory`, `total_code_files`, `code_size`, `scancode_notice`,
+    `scancode_version`, `files_count`, `total_error`, `scan_time`
+    """
     def __str__(self):
         return self.scan_type
 
-    # types of scans that can be applied
     scan_types = (
-        ('URL', 'URL'),
-        ('Local Scan', 'localscan'),
+        ('url', 'url'),
+        ('localscan', 'localscan'),
     )
 
-    scan_type = models.CharField(max_length=20, choices=scan_types, default='URL')
-    is_complete = models.BooleanField()
-
-class UserInfo(models.Model):
-    def __str__(self):
-        return self.user.username
-
-    user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
-    scan_info = models.ForeignKey(ScanInfo)
-
-
-class URLScanInfo(models.Model):
-    def __str__(self):
-        return self.URL
-
-    scan_info = models.ForeignKey(ScanInfo)
-    URL = models.URLField(max_length=2000)
-
-
-class LocalScanInfo(models.Model):
-    def __str__(self):
-        return self.folder_name
-
-    scan_info = models.ForeignKey(ScanInfo)
-    folder_name = models.CharField(max_length=200)
-
-
-class CodeInfo(models.Model):
-    def __str__(self):
-        return self.total_code_files
-
-    scan_info = models.ForeignKey(ScanInfo)
+    scan_type = models.CharField(max_length=20, choices=scan_types, default='url')
+    is_complete = models.BooleanField(default=False)
+    user = models.ForeignKey(User, blank=True, null=True)
+    URL = models.URLField(max_length=2000, blank=True, null=True)
+    scan_directory = models.CharField(max_length=200)
     total_code_files = models.IntegerField(null=True, blank=True)
     code_size = models.IntegerField(null=True, blank=True, default=0)
-
-
-# ScanResult will store the most basic results of the scan
-# Single result for each scan
-class ScanResult(models.Model):
-    def __str__(self):
-        return self.total_errors
-
-    code_info = models.ForeignKey(CodeInfo)
-    scanned_json_result = JSONField()
-    scanned_html_result = models.CharField(max_length=10000)
-    scancode_notice = models.CharField(max_length=2000)
-    scancode_version = models.CharField(max_length=200)
+    scancode_notice = models.CharField(max_length=2000, blank=True, null=True)
+    scancode_version = models.CharField(max_length=200, blank=True, null=True)
     files_count = models.IntegerField(null=True, blank=True, default=0)
     total_errors = models.IntegerField(null=True, blank=True, default=0)
     scan_time = models.IntegerField(null=True, blank=True, default=0)
```

Comment:
Would having a start_date and end_date (as Date/times types) be better? a scan time alone is not a great piece of data.